dcicutils/misc_utils.py

```diff
@@ -1173,6 +1173,41 @@ def find_association(data, **kwargs):
         raise ValueError("Got %s results when 1 was expected." % n)
 
 
+def keys_and_values_to_dict(keys_and_values: list, key_name: str = "Key", value_name: str = "Value") -> dict:
+    """
+    Transforms the given list of key/value objects, each containing a "Key" and "Value" property,
+    or alternately named via the key_name and/or value_name arguments, into a simple
+    dictionary of keys/values and returns this value. For example, given this:
+
+      [
+        { "Key": "env",
+          "Value": "prod"
+        },
+        { "Key": "aws:cloudformation:stack-name",
+          "Value": "c4-network-main-stack"
+        }
+      ]
+
+    This function would return this:
+
+      {
+        "env": "prod",
+        "aws:cloudformation:stack-name": "c4-network-main-stack"
+      }
+
+    :param keys_and_values: List of key/value objects as described above.
+    :param key_name: Name of the given key property in the given list of key/value objects; default to "Key".
+    :param value_name: Name of the given value property in the given list of key/value objects; default to "Value".
+    :returns: Dictionary of keys/values from given list of key/value object as described above.
+    """
+    result = {}
+    for item in keys_and_values:
+        key = item.get(key_name)
+        if key:
```

Comment:
I might just do something like this (untested). I don't know that I like tolerating value_name being missing. And I think it's good to mention situations that are not in proper form. Do we expect these might be in bad form for the intended first use? In general, though, the point of putting something like this in a library is so it has multiple uses, so I think tolerating bad data is just not good unless you make an argument that requests it.
```
result = {}
for item in keys_and_values:
    if key_name not in item:
        raise ValueError(f"Key {key_name} is not in {item}.")
    if value_name not in item:
        raise ValueError(f"Key {value_name} is not in {item}.")
    if key_name in result:
        raise ValueError(f"Key {key_name} is duplicated in {keys_and_values}.")
    result[item[key_name]] = result[item[value_name]]
return result
```
Sorry if I told you in the other PR you'd get away with using the code you had there in that very specialized case. Now that it's here, new scrutiny required. :)

Reply:
Yeah, sounds good, should we also check if the value of the key is a string (doesn't have to be I guess  - I was doing str on in just to be sure; and probably also check that given key_name and value_name are both (non-empty) strings.

Reply:
All set.